Source/Core/AI/WorkGivers/WorkGiver_Supervise.cs

```diff
@@ -11,21 +13,62 @@ internal class WorkGiver_Supervise : WorkGiver_Warden
         public override Job JobOnThing(Pawn pawn, Thing t, bool forced = false)
         {
             var prisoner = t as Pawn;
-            var need = prisoner?.needs.TryGetNeed<Need_Motivation>();
-            if (need == null || prisoner == null)
-                return null;
-            if (!ShouldTakeCareOfPrisoner(pawn, prisoner))
+            var mNeed = prisoner?.needs.TryGetNeed<Need_Motivation>();
+
+            if (pawn.IsPrisoner)
                 return null;
-            if (prisoner.Downed || !pawn.CanReserve(t, 1, -1, null, false) || !prisoner.Awake())
+
+            if (mNeed == null || prisoner == null || prisoner.Dead)
                 return null;
-            if (pawn.IsPrisoner)
+
+            if (!ShouldTakeCareOfPrisoner(pawn, prisoner))
                 return null;
-            var escapeTracker = EscapeTracker.Of(prisoner, true);
-            if (!PrisonLaborUtility.LaborEnabled(prisoner) && !escapeTracker.CanEscape)
+
+            if ((!prisoner.InBed() && prisoner.Downed) || !prisoner.Awake())
                 return null;
+
             if (PrisonLaborUtility.RecruitInLaborEnabled(prisoner))
                 return new Job(JobDefOf.PrisonerAttemptRecruit, t);
-            if ((!PrisonLaborUtility.WorkTime(prisoner) || !need.ShouldBeMotivated) && !escapeTracker.CanEscape)
+
+            if (prisoner.needs.food.CurLevelPercentage < 0.15 && !prisoner.Downed)
+            {
+                var curRoom = prisoner?.GetRoom();
+                var room = prisoner?.CurrentBed()?.GetRoom() ?? curRoom;
+
+                if (room != null && pawn.CanReserve(t, 1, -1, null, false))
+                {
+                    if (curRoom == null)
+                        goto CheckFood;
```

Comment:
hell no! Please create methods

Reply:
well right, have to make sure it is working first since something is messing the think tree.

Reply:
well with `goto` instruction code isnt ready to merge.